src/nylas-connection.ts

```diff
@@ -227,8 +230,18 @@ export default class NylasConnection {
 
   request(options: RequestOptions): Promise<any> {
     const req = this.newRequest(options);
+    let controller: AbortController;
+    let timeout: NodeJS.Timeout;
+
+    if (config.timeout && config.timeout !== 0) {
+      controller = new AbortController();
+      timeout = setTimeout(() => {
+        controller.abort();
+      }, 150);
+    }
+
     return new Promise<any>((resolve, reject) => {
-      return fetch(req)
+      return fetch(req, { signal: controller?.signal as AbortSignal })
```

Comment:
This should prevent you from type casting.
```suggestion
      return fetch(req, { ...(controller && { signal: controller.signal }) })
```